gateway-js/src/__tests__/integration/configuration.test.ts

```diff
@@ -138,6 +147,26 @@ describe('gateway configuration warnings', () => {
     // which triggers a different error that we're not testing for here.
     gateway = null;
   });
+
+  it('starts successfully using env variables for configuration', async () => {
+    cleanUp = mockedEnv({
+      APOLLO_GRAPH_VARIANT: 'env-config',
+      APOLLO_KEY: 'service:envtests'
```

Comment:
This is not expected to work in Apollo Server 3, which does not include an API key parser. You need to specify APOLLO_GRAPH_ID or APOLLO_GRAPH_REF.